src/main/java/com/exasol/adapter/dynamodb/documentpath/LinearDocumentPathWalker.java

```diff
@@ -0,0 +1,48 @@
+package com.exasol.adapter.dynamodb.documentpath;
+
+import com.exasol.adapter.dynamodb.documentnode.DocumentNode;
+
+/**
+ * This class is a simplified version of {@link DocumentPathWalker}. It has the limitation that it can only walk linear
+ * paths. That means that the paths must not contain {@link ArrayAllPathSegment}s.
```

Comment:
```suggestion
 * This class is a simplified version of {@link DocumentPathWalker}. 
 * 
 * It has the limitation that it can only walk linear paths. That is, the paths should not 
 * contain {@link ArrayAllPathSegment}s; otherwise it throws an exception.
```

Reply:
Okay, now read the method javadoc below, you can put the part with limitation and exception there on the method javadoc, and remove it from class javadoc here.